[CBRD-26060] log applier 모듈에 에러 매크로 추가

youngjun9072 opened the pull request:
http://jira.cubrid.org/browse/CBRD-26060
### Purpose

제어문에 조건식이 길어 파악하는데 어려움이 있는 코드 일부 리팩토링을 하였습니다.

### Implementation

IS_CONNECTION_ERROR, IS_LA_FLUSH_ERROR 두 매크로를 추가하였습니다.

IS_CONNECTION_ERROR는 네트워크 연결 과정에서 발생하는 에러 
ER_NET_CANT_CONNECT_SERVER 혹은 ER_OBJ_NO_CONNECT 인 경우를 의미하고

IS_LA_FLUSH_ERROR는 리플 객체를 플러시하는 과정에서 발생하는 에러
ER_LC_PARTIALLY_FAILED_TO_FLUSH 혹은 ER_LC_FAILED_TO_FLUSH_REPL_ITEMS를 의미합니다


### Remarks

* flush 는 다른 모듈에서도 발생할 수 있는 에러라고 생각되어 LA 키워드를 포함시켰습니다.
* 코드상 /* check connection error */는 매크로(혹은 기존 코드에서) 역할이 명확하기에 불필요한 주석이라고 생각되어 삭제했습니다.

youngjun9072 committed:
에러 매크로 추가

hyahong reviewed (approved):
LGTM

hornetmj commented on src/transaction/log_applier.c:
```diff
@@ -173,6 +173,12 @@
     } \
   while (0)
 
+#define IS_CONNECTION_ERROR(err) \
+  ((err) == ER_NET_CANT_CONNECT_SERVER || (err) == ER_OBJ_NO_CONNECT)
+
+#define IS_LA_FLUSH_ERROR(err) \
```
log_applier.c 파일에서만 사용되며, 이 파일 매크로는 LA_라는 prefix를 갖습니다. 일관성 있게 적용하면 좋을 것 같습니다.

```suggestion
#define LA_IS_CONNECTION_ERROR(err) \
  ((err) == ER_NET_CANT_CONNECT_SERVER || (err) == ER_OBJ_NO_CONNECT)

#define LA_IS_FLUSH_ERROR(err) \
```

youngjun9072 replied:
@hornetmj 
error == ER_NET_CANT_CONNECT_SERVER || error == ER_OBJ_NO_CONNECT 이 execute에도 존재해 재사용 가능성이 있어보여 connection 에러는 la키워드를 붙이지는 않았습니다.
execute쪽 코드를 저희팀에서 건드려도될까요?

hornetmj replied:
1. 이슈 처리를 위해 큐브리드 어떤 코드도 수정하셔도 무방합니다. 수정 시 code owner가 있기 때문에 자동 리뷰어로 등록될 것 입니다.
2. 전역 매크로로 수정하는 것보다 모듈 내에서 사용되는 매크로로 한정하시는 것이 추후 유지보수 하기 좋습니다. 다만, 단순 매크로이기 때문에 error_manager.h에 ER_IS_CONNECTION_ERROR() 등으로 등록 검토해 보시길 바랍니다.

youngjun9072 replied:
@hornetmj  error_manager 로 옮기려보니 이미 ER_IS_SERVER_DOWN_ERROR가 있지만 ER_NET_CANT_CONNECT_SERVER가 빠진 상태입니다.
applylogdb 입장에서보면 클라이언트가 서버에 연결하는 시점에 발생하는 에러라 ER_IS_SERVER_DOWN_ERROR 매크로에 해당 에러가 추가되도 괜찮다고 생각해 기존 매크로에 에러를 추가했고, 활용하는 방안을 택했습니다.

추가적으로 말씀하신대로 IS_LA_FLUSH_ERROR 는 LA_IS_FLUSH_ERROR로 변경하였습니다.

youngjun9072 committed:
리뷰 반영

- error manager 에 ER_NET_CANT_CONNECT_SERVER 에러 추가
- IS_CONNECTION_ERROR 에러를 ER_NET_CANT_CONNECT_SERVER 로 대체
- IS_LA_FLUSH_ERROR 를 LA_IS_FLUSH_ERROR로 변경

youngjun9072 commented:
/run sql medium

youngjun9072 commented:
/run sql medium

youngjun9072 committed:
오류 수정

youngjun9072 commented:
/run sql medium
